### ISSUE TITLE
Bad Credentials again (Connexoon)

### ISSUE THREAD

**Issue by posul** [dubocr/homebridge-tahoma#118]
Hi,

Thanks for all your work on this plugin.

I'm getting the Bad Credentials error that was reported some time ago in Issue #65. I have a Somfy account out of Australia - https://www.somfy.com.au.  Some months ago when I came across the Bad Credentials error, I saw your advice on issue #65. Following that I set my service as "Connexoon RTS" and I was able to get past the error. I've since setup a new Homebridge Server and I'm getting that same Bad Credentials error again. I've checked https://dev.duboc.pro/tools/setup and there also getting "Bad Credentials" regardless of which server I use. My config and Logs below.

Hoping you can offer some help.
Thanks
Paul

My config;
   {
      "platform": "Tahoma",
      "name": "Awning",
      "user": "XXXXXX",
      "password": "XXXXXX",
      "service": "Connexoon RTS"
    }
 

My logs;
0|homebridge  | [2019-8-4 16:17:51] [Awning] RESP : {"errorCode":"AUTHENTICATION_ERROR","error":"Bad credentials"}
0|homebridge  | [2019-8-4 16:17:51] [Awning] Login fail: Bad credentials
0|homebridge  | [2019-8-4 16:17:51] [Awning] Bad credentials
0|homebridge  | [2019-8-4 16:17:52] [Awning] RESP : {"errorCode":"AUTHENTICATION_ERROR","error":"Bad credentials"}
0|homebridge  | [2019-8-4 16:17:52] [Awning] Login fail: Bad credentials
0|homebridge  | TypeError: Cannot read property 'length' of null
0|homebridge  |     at Server.<anonymous> (/usr/local/lib/node_modules/homebridge/lib/server.js:407:44)
0|homebridge  |     at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/lib/util/once.js:16:19
0|homebridge  |     at /home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/index.js:93:6
0|homebridge  |     at TahomaPlatform.<anonymous> (/home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/index.js:194:5)
0|homebridge  |     at /home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/overkiz-api.js:189:4
0|homebridge  |     at Request._callback (/home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/overkiz-api.js:262:25)
0|homebridge  |     at Request.self.callback (/home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/node_modules/request/request.js:185:22)
0|homebridge  |     at emitTwo (events.js:126:13)
0|homebridge  |     at Request.emit (events.js:214:7)
0|homebridge  |     at Request.<anonymous> (/home/pi/.nvm/versions/node/v8.16.0/lib/node_modules/homebridge-tahoma/node_modules/request/request.js:1161:10)

**Comment by dubocr** [comment]
Hi,

Have you tried again 'Connexoon' ?
Can you log at https://dev.duboc.pro/tools/setup
and try each server and tell me if one of the match.

**Comment by posul** [comment]
Hi, 
I tried again and 'Connexoon RTS' worked this time. Running well now.
Thanks again.

**Comment by bowana** [comment]
Hi, has anything changed on this recently?  My credentials haven't change and yet I am having authentication issues again.  I've tried the API tool and also have "Bad credentials".  Anyone else also having this issue?

**Comment by yenoiwesa** [comment]
@bowana I was impacted by the same issue (I am also based in Australia). I found in the Connexoon RTS app that there was a scheduled maintenance planned for the next day when the authentication issue started. I suspect it might have been related.

However I was surprised that the iOS and Android apps would still manage to login when the homebridge plugin wouldn't. I checked the requests sent by the Android app during the login sequence, and they actually use the somfy.com domain to generate an access token, that they then pass to each subsequent request to the connexoon server for commands.

I am maintaining [homebridge-connexoon](https://github.com/dubocr/homebridge-connexoon), and just released a new version of it with the authentication method swapped for the somfy.com based access token.
Hopefully that fixes the intermittent issues of Bad Credentials error. Time will tell...
